### DISCUSSION TITLE
Unable to build openssl on HP NONSTOP guardian platform.

### DISCUSSION THREAD

**Discussion by prasant-patra** [openssl/openssl#23107] [Q&A] [answered]
Hello all,

Recently we are trying to upgrade one of our application code to use openssl 3.0 in HP Nonstop OS platform.
Thought we can easily use openssl library available at https://ituglib.connect-community.org/apps/Ituglib site.
But that seems to give us some linking issue because it was built for OSS environment in Nonstop.
Our application runs on **GUARDIAN** environment, Hence we have to build openssl and then can link to our application.

In openssl documentation it has help file for nonstop **NOTES-NONSTOP.md**. Currently running configure script with below options from **OSS** terminal.
./**Configure nonstop-nsx_g_tandem** --prefix=/home/openssl/prasant/openssl_3.0_lib --openssldir=/home/openssl/prasant/openssl_3.0_lib/ssl no-shared no-threads no-makedepend --with-rand-seed=os
After that when I use make command on above generated Makefile, It gives below error. As we know kill api is undefined in guardian environment.

Can someone tell me why it tries to compile non guardian code when we have set configure option nonstop-nsx_g_tandem ?
Am I using correct configure options to build for guardian environment?
Appreciate your help to complete openssl build on nonstop platform!

c99: running /usr/cmplr/ccomfe --c99 -ZO2 -Zg3 -ZS/tmp/t738197510AABFHJ.T -ZtG -
ZR8 -m -Zr -ZT -D_TANDEM_SOURCE -ZNW203 -ZNW220 -ZNW272 -ZNW734 -ZNW770 -ZNW1506
 -Zz -I. -Iinclude -Iapps/include -I/usr/include -DOPENSSL_PIC -DOPENSSLDIR="/ho
me/openssl/prasant/openssl_3.0_lib/ssl" -DENGINESDIR="/home/openssl/prasant/open
ssl_3.0_lib/lib/engines-3" -DMODULESDIR="/home/openssl/prasant/openssl_3.0_lib/l
ib/ossl-modules" -DOPENSSL_BUILDING_OPENSSL -DOPENSSL_VPROC= -D_XOPEN_SOURCE -D_
XOPEN_SOURCE_EXTENDED=1 -D_TANDEM_SOURCE -DB_ENDIAN -DNO_GETPID -D_TANDEM_ARCH=3
 -DNDEBUG apps/lib/http_server.c
              (void)kill(kidpids[i], SIGTERM);
                    ^
"/home/openssl/prasant/openssl-3.0.12/apps/lib/http_server.c", line 97: error(11
4):
          identifier "kill" is undefined

1 error detected in the compilation of "apps/lib/http_server.c".
c99: /usr/cmplr/ccomfe exited, returning 2.
c99: Exiting with status 1; 0 error(s).

Thanks!
Prasant

**Comment by mattcaswell** [comment]
@rsbecker

**Comment by rsbeckerca** [comment, marked as answer]
@prasant-patra Guardian builds are not yet supported by the ITUGLIB team. I am the community maintainer for NonStop for OSS builds. We have had requests to do Guardian but there are serious limits on what is possible currently. I know of one group who was successful, but they have not yet shared their port. The `nonstop-nsx_g_tandem` is a holdover from 1.1.1 that we have tried to get to work. It is a work in progress more than anything else and not supported here. We may get this to work eventually but definitely **not** for the 3.0.x or 3.1.x series - remember, no one is paying the ITUGLIB team to do this, so day jobs get in the way.

**Reply by prasant-patra** [reply]
Thanks @rsbeckerca for your quick response. Now I understand current situation. 
As you have mentioned in past you have seen one group have successfully ported, some contact would be very helpful. I will try understanding which openssl version they have compiled.

**Reply by rsbeckerca** [reply]
@prasant-patra I am sorry but I cannot disclose the organization that has successfully ported the code for NDA reasons.

**Comment by rsbeckerca** [comment]
@prasant-patra If you need further contact/requests for ITUGLIB, please use the [LinkedIn ITUGLIB group](https://www.linkedin.com/groups/14360197/).

**Reply by prasant-patra** [reply]
Thanks for the ITUGLIB group link.

**Reply by prasant-patra** [reply]
Hi @rsbeckerca ,
It seems openssl porting is not simple to nonstop guardian.
Is there any other openssl alternative TLS toolkit you can suggest me for nonstop guardian platform?

Many thanks in advance for your attention!

**Reply by shobhanjalir** [reply]
Any updates on this? Has anyone successfully ported openssl later versions to nonstop guardian?

**Reply by rsbeckerca** [reply]
> Any updates on this? Has anyone successfully ported openssl later versions to nonstop guardian?

There are two ports for OpenSSL to Guardian. Both are on the SUT One is SSL Toolkit (3.0.14) the other is NonStop SSL. I am not sure what version that currently uses.

4.0.0 and 3.5 have not been ported, but I think those are in progress.

**Comment by rsbeckerca** [comment]
@prasant-patra Not as far as I know.
